stablehlo/reference/Ops.cpp

```diff
@@ -541,6 +546,23 @@ Tensor evalLogisticOp(const Tensor &operand, TensorType resultType) {
   return result;
 }
 
+Tensor evalMapOp(ArrayRef<Tensor> inputs, Region &computation, Scope &scope,
+                 TensorType resultType) {
+  Tensor result(resultType);
+  for (auto inputIt = inputs[0].index_begin(); inputIt != inputs[0].index_end();
+       ++inputIt) {
+    SmallVector<Tensor> args;
+    for (size_t i = 0; i < inputs.size(); ++i) {
+      auto tensor =
```

Comment:
No need to extract an tensor element to get the element type of the tensor.
```
Tensor arg(RankedTensorType::get({}, inputs[i].getElementType());
```

Even better: IIRC there is a PR where we talked about introducing a constructor like 
```
 explicit Tensor(TensorType, const Element &);
```

We should use it here as well.

Reply:
Thanks for the suggestion. I also recall discussing this somewhere but haven't find which PR it was (unfortunately GH doesn't have a great search engine).